lib/target.py

```diff
@@ -242,20 +328,30 @@ def __init__(self, triple):
             self.sdk = OSType.Win32
             self.dynamic_library_suffix = ".dll"
             self.executable_suffix = ".exe"
+        elif "darwin" in triple:
+            self.sdk = OSType.MacOSX
+        else:
+            print("Unknown platform")
+
         self.triple = triple
+
         comps = triple.split('-')
-        ArchType.from_string(comps[0])
+        self.arch = ArchType.from_string(comps[0])
 
     @staticmethod
     def default():
-        triple = platform.machine() + "-"
+        arch = ArchType.from_string(platform.machine())
+        triple = ArchType.to_string(arch)
         if platform.system() == "Linux":
-            triple += "linux-gnu"
+            if arch == ArchType.arm:
+                triple += "-linux-gnueabihf"
```

Comment:
what about "-linux-armeabi" or "-linux-android" etc? I know these are other handled cases in llvm's triple calculations